src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/actions/GenerateTestsActionClass.kt

```diff
@@ -22,14 +22,13 @@ class GenerateTestsActionClass : AnAction() {
      * @param e an action event that contains useful information and corresponds to the action invoked by the user
      */
     override fun actionPerformed(e: AnActionEvent) {
-        // Return if EvoSuite is already running
-        val project = e.project ?: return
-        val runnerService = project.service<RunnerService>()
-        if (runnerService.verifyIsRunning()) return
-
```

Comment:
Why doesn't it make sense to first check for running and parseability before checking the cache?
Won't it be unnecessary work to update the cache if the class is not parsable or if evo is running?

Reply:
This has been moved below in code, because I need to make use of the `psiFile`.